### PR TITLE
Set cesium base url in Terria constructor.

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

In Terria v7 we used to set cesium base url. It was either passed in as an option to the Terria constructor or set by default from `Terria.baseUrl` value. Setting cesium base url ensures that cesium correctly resolves the URLs to its static assets (like the bing maps logo in this issue - https://github.com/TerriaJS/terriajs/issues/6473).

This PR ports that bit of code to set cesium base URL from `v7`.
  
### Test me
  
- Open the CI for main branch - http://ci.terria.io/main/
- Switch to any bing map provider and click the `Attributions` link at the bottom of the map
- Observe that the Bing Map logo is broken
- Now Open the CI for this branch - http://ci.terria.io/set-cesium-baseurl/
- Repeat
- Observe that the Bing map logo is visible

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated relevant documentation in `doc/`.
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Core/ensurePrefix.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Ensures that the given `str` ends with the given `char`.
+ *
+ */
+export default function ensurePrefix(str: string, char: string): string {
```
Comment (tephenavies) [inline lib/Core/ensurePrefix.ts]: Should be named `ensureSuffix`

File: lib/Models/Terria.ts [hunk 2]
```diff
@@ -620,14 +635,18 @@ export default class Terria {
...
-      if (options.baseUrl.lastIndexOf("/") !== options.baseUrl.length - 1) {
-        this.baseUrl = options.baseUrl + "/";
-      } else {
-        this.baseUrl = options.baseUrl;
-      }
+      this.baseUrl = ensurePrefix(options.baseUrl, "/");
     }
 
+    this.cesiumBaseUrl = ensurePrefix(
+      options.cesiumBaseUrl ?? `${this.baseUrl}build/Cesium/build`,
```
Comment (tephenavies) [inline lib/Models/Terria.ts]:
```suggestion
      options.cesiumBaseUrl ?? `${this.baseUrl}build/Cesium/build/`,
```

Provide the default in the preferred form.

**Review by tephenavies — CHANGES_REQUESTED** [review]

File: lib/ReactViews/Credits/DataAttribution/DataAttributionModal.tsx [hunk 3]
```diff
@@ -35,17 +47,14 @@ const DataAttributionBox = styled(Box).attrs({
...
-
-    img {
-      height: 19px;
-      vertical-align: middle;
-    }
+  /* Default cesium bing map logo is white on transparent which is rendered invisible
+     on our modal with white background. This rule forces the background color of the
+      bing imagery to grey so that it is visible.
+  */
+  ${AttributionText} img[title="Bing Imagery"] {
```
Comment (mwu2018) [inline lib/ReactViews/Credits/DataAttribution/DataAttributionModal.tsx:55]: Making changes as below will produce better result than grey background.
```
  ${AttributionText} img[title="Bing Imagery"] {
    -webkit-filter: invert(1);
    filter: invert(1);
  }
```

Ref: https://stackoverflow.com/questions/13325798/invert-colors-of-an-image-in-css-or-javascript
Test: http://ci.terria.io/mike-testing/

Comment (tephenavies) [reply lib/ReactViews/Credits/DataAttribution/DataAttributionModal.tsx:55]: `-webkit-filter` vendor prefixing is not needed. styled-components will add appropriate vendor prefixes for you.

Comment (na9da) [reply lib/ReactViews/Credits/DataAttribution/DataAttributionModal.tsx:55]: Changed to `filter` :heart:

File: lib/Core/ensurePrefix.ts (hunk 1 above)
Comment (na9da) [reply lib/Core/ensurePrefix.ts]: Renamed :grimacing:

**Review by tephenavies — APPROVED** [review]

**Pull request merged**